packages/fast-usdc/src/type-guards.js

```diff
@@ -21,7 +21,13 @@ export const makeProposalShapes = ({ PoolShares, USDC }) => {
   /** @type {TypedPattern<USDCProposalShapes['deposit']>} */
   const deposit = M.splitRecord(
     { give: { USDC: makeNatAmountShape(USDC, 1n) } },
-    { want: M.splitRecord({}, { PoolShare: makeNatAmountShape(PoolShares) }) },
+    {
+      want: M.splitRecord(
+        {},
+        { PoolShare: makeNatAmountShape(PoolShares) },
+        {},
+      ),
+    },
```

Comment:
Good find. I wonder if this should just be:
```js
{ want: { PoolShare: makeNatAmountShape(PoolShares) } },
```

But I suppose someone could deposit without wanting shares in return?

Reply:
I'm struggling to understand the change. It adds a 3rd arg to `splitRecord(...)`? What does that do? (_looking it up..._)

Reply:
>  It adds a 3rd arg to splitRecord(...)? What does that do? 

I think `M.splitRecord({}, {PoolShares: ... }, {});` says:
1. can be an empty object
2. if PoolShares is present, the value must be a USDC amount shape
**3. no other keys allowed**

Reply:
OK... I see `rest` in [splitRecord](https://endojs.github.io/endo/interfaces/_endo_patterns.PatternMatchers.html#splitRecord) docs.

I wonder how many other places I have missed this.

Reply:
yes, 0 pool shares would work.

I was trying to avoid burdening callers with getting access to the PoolShares brand when they are OK relying on the contract to figure it out... but that's really not a normal scenario. In normal usage, clients should _not_ rely on the contract to behave nicely. That's sort of the whole point of offer safety.

Let's be sure to keep the type and the typeguard in sync. If we make the `{ want: { PoolShare: ... }}` part required, we can probably simplify some code that consumes it.

Reply:
> If we make the { want: { PoolShare: ... }} part required, we can probably simplify some code that consumes it.

Done